File: src/DutyFree.Entity/ValueType/User/Wallet.cs
namespace DutyFree.Entity.ValueType.User
{
    using Platform.Extension;
    using System.Collections.Generic;
    using System.Linq;

    public class Wallet
    {
        //钱包余额
        public double GetTotal()
        {
            double inNumber = 0;
            double outNumber = 0;

            if (!InItems.IsEmpty())
            {
                inNumber = InItems.Sum(o => o.Number);
            }

            if (!OutItems.IsEmpty())
            {
                outNumber = OutItems.Sum(o => o.Number);
            }

            return inNumber - outNumber;
        }

        //获取记录
        public List<WalletItem> InItems { get; set; }

        //使用记录
        public List<WalletItem> OutItems { get; set; }

        public Wallet()
        {
            this.InItems = new List<WalletItem>();
            this.OutItems = new List<WalletItem>();
        }
    }
}

File: bomeiboyou/src/entity/wallet.ts
import { WalletItem } from './wallet-item';

export class Wallet {
    public constructor(public inItems: WalletItem[], public outItems: WalletItem[]) { }

    public getTotal(): number {
        let inTotal = 0;
        let outTotal = 0;

        if (this.inItems.length > 0) {
            inTotal = this.inItems.map(o => o.number).reduce((pre, cur) => pre + cur);
        }

        if (this.outItems.length > 0) {
            outTotal = this.outItems.map(o => o.number).reduce((pre, cur) => pre + cur);
        }

        return inTotal - outTotal;
    }
}